### PR TITLE
Pull Request: Submit vignette for Feature naming_metrics 20240916

### PR DESCRIPTION
**Pull request by andreabr-dixie**
Pull Request: Submit vignette for Feature naming_metrics 20240916

Description
This pull requests that we add a new vignette to utDataStoR - naming_metrics. This vignette describes the naming convention of metrics in the utMetrics package and gives some examples. The utMetrics package is private, so placing this vignette in utDataStoR allows it to be placed in a public bookdown. The vignette describes how a metric name is derived, the metrics components, and gives two examples

Changes Made
Added a new vignette titled "naming_metrics"
Includes the following:
A description of how a metric name is derived
Shows the metric components
Gives 2 examples of metrics and how they are built (and provides a diagram)
Provides a list of Computations
Provides a list of Computations and Partitions
Provides a list of Computations, Partitions and Segments

Motivation and Context
The goal of this vignette is to help users understand the metric naming convention, show examples of how the metric components create the metrics names, and show what metrics are available.

How Has This Been Tested?
The vignette has been reviewed for clarity and accuracy by the Office of Institutional Effectiveness.
All code examples have been tested to ensure they produce the expected results.

Styling
The styling of this vignette has been reviewed and approved by the Office of Institutional Effectiveness.

Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Checklist

- [x] My vignette follows the style guidelines provided in the README for this repository
- [x] I have performed a self-review of my own code, including knitting the vignette
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have reviewed my .gitignore file to ensure the exclusion of items we do not want included in a commit history
- [x] My vignette does not include any sensitive data

Additional Notes

### CODE REVIEW

**Comment by andreabr-dixie** [comment]
@MattNickodemus please review this pull request today for utmetrics vignettes as discussed in morning huddle.

File: vignettes/naming_metrics.Rmd [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+library(DT)
+```
+
+## Metric Names
+Metric names are designed to give a clear understanding of the metric.
+Metric names are derived from components of the Metric.
+
+- Season (if applicable)
+- Computation
+- by
```
Comment (MattNickodemus) [inline vignettes/naming_metrics.Rmd:37]: I think we want to use partition here.

**Review by MattNickodemus — APPROVED** [review]
This is excellent work @andreabr-dixie

**Comment by MattNickodemus** [comment]
@cdemkegit This is ready to merge to main whenever you are ready.

**Comment by cdemkegit** [comment]
I'm merging this to dev

**Pull request merged**